backend/src/services/git.ts

```diff
@@ -75,9 +75,20 @@ const setupGitConfig = (username: string, email: string, repositoryName: string)
 const gitCheckout = async (git:SimpleGit, branchName: string) => {
   const sanitizedBranchName = sanitizeBranchName(branchName)
   await validateBranchName(sanitizedBranchName)
-  await git.checkout([sanitizedBranchName])
+  
+  if (await branchExists(git, sanitizedBranchName)) {
+    await git.checkout([sanitizedBranchName])
+    return
+  }
+  
+  await git.checkout(['-b', sanitizedBranchName])
 }
 
+const branchExists = async (git:SimpleGit, branchName: string):Promise<boolean> => {
+  const branchSummaryResult = await git.branch(['--list', branchName])
+  return Object.keys(branchSummaryResult.branches).find(branch => branch === branchName) ? true : false
```

Comment:
Javascript arrays has method some() that returns true or false if element is found. We should use it instead of find()